django/contrib/auth/hashers.py

```diff
@@ -201,6 +202,23 @@ def encode(self, password, salt):
         """
         raise NotImplementedError('subclasses of BasePasswordHasher must provide an encode() method')
 
+    def decode(self, encoded):
+        """
```

Comment:
I am wondering currently if we cannot make this `NotImplementedError` already. I mean there is no *generic* code (at least not yet) which calls this method. In the future `must_update` could call it to check for a change in `salt_len` -- but that will also only happen once the hasher call `super().must_update`. What do you think @felixxm?

Reply:
Agreed, we can add a deprecation warning later in `must_update()`.